### DISCUSSION TITLE
Demo: Rootless Podman + MariaDB + socket activation

### DISCUSSION THREAD

**Discussion by eriksjolund** [containers/podman#13167] [General]
I made a small demo about running MariaDB with rootless Podman and socket activation
https://github.com/eriksjolund/mariadb-podman-socket-activation

I tested adding  _--network none_ for both TCP and UNIX sockets
and it still worked to connect with the MariaDB client from the host.

Templated systemd user services work really well with UNIX sockets because you can
"auto-generate" a new path for a UNIX socket with the configuration
```
ListenStream=%h/mariadb-socket.%i
```
(`%h` expands to the home directory path. `%i` expands to the instance name)

When using TCP sockets, I guess you need to provide one systemd override configuration file
per instance, to be able to give each instance its own TCP port number.

**Comment by rhatdan** [comment]
Interested in recording it and publishing it?

**Comment by eriksjolund** [comment]
I could make a tweet. I'll just polish the text content somewhat first.

Another thing, I tested some more and it seems __--security-opt label=disable__ is needed for UNIX sockets but not for TCP sockets. Is that result according to expectations?

**Reply by rhatdan** [reply]
Yes SELinux blocks interprocess communications, but does not block network communications. since it can not tell the difference between the container connecting to another host or to the same host.

**Reply by eriksjolund** [reply]
Thanks, for explaining.

**Reply by eriksjolund** [reply]
Tweet: https://twitter.com/eriksjolundcomp/status/1491162016429768704
